source/textInfos/__init__.py

```diff
@@ -363,9 +369,11 @@ def updateSelection(self):
 		"""Moves the selection (usually the system caret) to the position of this text info object"""
 		raise NotImplementedError
 
+	_abstract_bookmark = True
 	def _get_bookmark(self):

```

Comment:
Why can't this also just use `@abstractmethod`?

Reply:
Because that would mark _get_bookmark as an abstract method. Imagine the following subclass:

```
class Test(textInfos.TextInfo):
    Bookmark = None
```

Here, bookmark is overridden at the class level and therefore should no longer be abstract. However, when we mark _get_bookmark as an abstract method, the subclass wouldn't instantiate because _get_bookmark is still abstract. To summarize, using abstractmethod for magic _get_ and _set_ methods will make it impossible to override a property at the subclass by means of the method explained above.

I hope this clears up things a little bit.